staging/src/k8s.io/kubectl/pkg/cmd/run/run.go

```diff
@@ -173,13 +169,9 @@ func NewCmdRun(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Co
 
 func addRunFlags(cmd *cobra.Command, opt *RunOptions) {
 	cmdutil.AddDryRunFlag(cmd)
-	cmd.Flags().StringVar(&opt.Generator, "generator", opt.Generator, i18n.T("The name of the API generator to use, see http://kubernetes.io/docs/user-guide/kubectl-conventions/#generators for a list."))
-	cmd.Flags().MarkDeprecated("generator", "has no effect and will be removed in the future.")
```

Comment:
We can't remove then, yet. They were just recently deprecated and that means we'll need to keep it around for quite a while, even if they don't do anything.

Reply:
Ok, that was my original intention , but then I thought changing to fail would be the same as removing them basically, but I understand.  I will change them to fail of you give an "unsupported value" and update this PR.

Reply:
OK, PR has been updated to prevent the situation where you try to use run to create a deployment (or other resource) and get a pod instead. PTAL.